packages/controller-utils/src/create-service-policy.ts

```diff
@@ -130,6 +131,18 @@ export const DEFAULT_CIRCUIT_BREAK_DURATION = 30 * 60 * 1000;
  */
 export const DEFAULT_DEGRADED_THRESHOLD = 5_000;
 
+const isServiceFailure = (error: unknown) => {
+  if (error && hasProperty(error, 'code')) {
+    const { code } = error;
+    // Don't consider -32601 (method not found) and -32005 (rate limit exceeded) as service failures
+    if (code === -32601 || code === -32005) {
```

Comment:
Why these two codes specifically?

Looking at the JSON-RPC spec, it appears that all codes except -32603 represent client errors: https://www.jsonrpc.org/specification

That's the only code I can see in https://www.npmjs.com/package/eth-rpc-errors that represents a server error as well, as far as I can tell.

Reply:
I’ve updated the function to treat only numeric error codes with the value -32603 as service failures.